examples/Example12/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+# SPDX-FileCopyrightText: 2021 CERN
+# SPDX-License-Identifier: Apache-2.0
+
+if(NOT TARGET G4HepEm::g4HepEm)
+  message(STATUS "Disabling Example12 (needs G4HepEm)")
+  return()
+endif()
+
+if(Geant4_FOUND)
+  if(NOT Geant4_gdml_FOUND)
+    message(STATUS "Disabling Example12 (needs Geant4 with GDML support)")
+    return()
+  endif()
+else()
+  message(STATUS "Disabling Example12 (needs Geant4)")
+  return()
+endif()
+
+# Needs VecGeom::vgdml
+if(NOT TARGET VecGeom::vgdml)
+  message(STATUS "Disabling Example12 (requires VecGeom with GDML support)")
+  return()
+endif()
+
+# Example12 is the AdePT demo example using material cuts as defined in the input gdml file
+add_executable(Example12
+  Example12.cpp
+  Example12.cu
```

Comment:
Lower-case `example12` please for consistency - I just spend too much of my time checking and re-checking why I have no binary `example12`...